frontend/packages/ux-editor/src/containers/DesignView/AddItem/useSearchComponent/useSearchComponent.ts

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState, type ChangeEvent, type KeyboardEvent } from 'react';
+import { useDebounce } from '@studio/hooks';
+import type { KeyValuePairs } from 'app-shared/types/KeyValuePairs';
+import type { IToolbarElement } from '../../../../types/global';
+
+const getAllLabelWords = (toolbarItem: IToolbarElement, t: (key: string) => string): string[] => {
+  const translatedLabel = t('ux_editor.component_title.' + toolbarItem.type).toLowerCase();
+  const englishLabel = toolbarItem.label.toLowerCase();
+  return [...translatedLabel.split(/\s+/), ...englishLabel.split(/\s+/)];
+};
+
+const filterToolbarItems = (
+  items: IToolbarElement[],
+  search: string,
+  t: (key: string) => string,
+): IToolbarElement[] => {
+  if (!search) return items;
+  return items.filter((toolbarItem) =>
+    getAllLabelWords(toolbarItem, t).some((word) => word.startsWith(search)),
+  );
+};
+
+type UseSearchComponentProps = {
+  availableComponents: KeyValuePairs<IToolbarElement[]>;
+  disableDebounce?: boolean;
+  t: (key: string) => string;
+};
+
+export function useSearchComponent({
+  availableComponents,
+  disableDebounce,
+  t,
+}: UseSearchComponentProps) {
+  const [searchText, setSearchText] = useState('');
+  const [debouncedSearchText, setDebouncedSearchText] = useState('');
+  const { debounce } = useDebounce({ debounceTimeInMs: disableDebounce ? 1 : 500 });
```

Comment:
What are the values `1` and `500` 🤔 Can we move them into constants with meaningful names? 😄